Runtime/McPublisher.cs

```diff
@@ -441,6 +446,79 @@ public void SetVideoSource(Camera source, StreamSize resolution = null)
             }
         }
 
+        /// <summary>
+        /// Set the video source for capture. Call with null as the source to remove
+        /// the currently set video source.
+        /// </summary>
+        /// <param name="source">The Target Render Texture source to capture from</param>
+        /// <param name="resolution">The capturing resolution</param>
+        public void SetVideoSource(RenderTexture source, StreamSize resolution = null)
```

Comment:
Can we refactor the internals of the two overloads of SetVideoSource as they are quite similar. I'd suggest splitting the internals into separate functions to reduce repetition

Reply:
refactored code to avoid duplicate code.